core/src/main/java/com/linecorp/armeria/common/multipart/MultipartFile.java

```diff
@@ -54,7 +57,49 @@ static MultipartFile of(String name, String filename, Path path) {
         requireNonNull(name, "name");
         requireNonNull(filename, "filename");
         requireNonNull(path, "path");
-        return new DefaultMultipartFile(name, filename, path);
+        return of(name, filename, path, HttpHeaders.of());
+    }
+
+    /**
+     * Creates a new {@link MultipartFile}.
+     * @param name the name parameter of the {@code "content-disposition"}
+     * @param filename the filename parameter of the {@code "content-disposition"}
+     *                 header.
+     * @param file the file that stores the {@link BodyPart#content()}.
+     * @param headers HTTP part headers.
+     */
+    static MultipartFile of(String name, String filename, File file, HttpHeaders headers) {
+        requireNonNull(name, "name");
+        requireNonNull(filename, "filename");
+        requireNonNull(file, "file");
+        requireNonNull(headers, "headers");
```

Comment:
We could delegate null checks.
```suggestion
        requireNonNull(file, "file");
```